agent/lib/src/api/model.rs

```diff
@@ -0,0 +1,187 @@
+use serde::{Deserialize, Serialize};
+
+
+/// Represents a file to be included in the workspace
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub struct FileModel {
+    /// Name of the file, paths relative to the workspace
+    pub filename: String,
+    /// Content of the file
+    pub content: String,
+}
+
+/// Identifies the type of the message
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq)]
+pub enum Type {
+    /// Status message to indicate that the agent is ready
+    Status,
+    /// Request message
+    Request,
+    /// Response message answering to a request message
+    Response,
+}
```

Comment:
I don't think `Type`is checked or used anywhere, except for decorating your messages with that information. As explained in https://github.com/virt-do/lambdo/pull/12#discussion_r1165096715, you will have to create a new structure (as opposed to extending a payload enum) for any new message that you'd want to extend your protocol with. So you don't need this `Type` enum and you should get rid of it.